src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -575,36 +575,41 @@ private function presentTags(array $localizedTagsList): array
     /**
      * @param ProductForEditing $productForEditing
      *
-     * @return array<string, int|array<int, int|array<string, string|int>>>
+     * @return array<string, array<int|string, array<int|string, int|string|null>|int>>
```

Comment:
Looks like you already know the schema of this array, consider using an array shape:
```suggestion
     * @return array{suppliers: array{default_supplier_id: int, supplier_ids: int[]}, product_suppliers: array<int, array{supplier_id: int, supplier_name: string, product_supplier_id: int, price_tax_excluded: string, reference: string, currency_id: int, combination_id: int}>}
```